app/assets/javascripts/Components/Result/result.jsx

```diff
@@ -43,12 +50,21 @@ class Result extends React.Component {
     });
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      this.state.props.submission_id !== prevState.props.submission_id ||
```

Comment:
Let's actually only check for changing `result_id`. The way our data is set up, if the submission id changes then the result id will also change.